### PR TITLE
Scores support

### PR DESCRIPTION
**Pull request by regisoliveira**
`SearchResultDecoder` crashes when using `io.github.dengliming.redismodule.redisearch.search.SearchOptions#withScores` equals to `true`, since the response we get from `FT.SEARCH` contains the score value just after the document key.

We will calculate the `document` size using `parts` data. After them, we use that size to iterate over the `parts` and build a `Document` with it's  Key a Payload (previous behavior) or Key, Score and Payload (new behavior).

It would be better if we could access the command's parameters inside the `SearchResultDecoder` to assure what kind of response we are dealing with.

Bellow there are 2 images that shows the parts differences without and with `withScores` parameter.

Captura de Tela 2021-04-17 às 21 38 44
Captura de Tela 2021-04-17 às 21 37 13

### CODE REVIEW

File: redisearch/src/main/java/io/github/dengliming/redismodule/redisearch/protocol/decoder/SearchResultDecoder.java [hunk 1]
```diff
@@ -30,24 +31,57 @@
...
+
+	@Override
+	public SearchResult decode(List<Object> parts, State state) {
+
+		int documentSize = 0;
+		Long total = (Long) parts.get(0);
+		boolean noContent = total == parts.size() + 1;
+
+		// Calculates Document Size searching for the first LinkedHashMap on parts...
+		// It would be better if I could access the commands parameters to check if we are dealing with the Score values...
```
Comment (dengliming) [inline redisearch/src/main/java/io/github/dengliming/redismodule/redisearch/protocol/decoder/SearchResultDecoder.java:47]: Why not?

```
// io.github.dengliming.redismodule.redisearch.protocol.RedisCommands
RedisCommand FT_SEARCH = new RedisCommand<>("FT.SEARCH", new ListMultiDecoder2(new SearchResultDecoder(), new StringMapInfoDecoder()));

RedisCommand FT_SEARCH_WITH_SCORES = new RedisCommand<>("FT.SEARCH", new ListMultiDecoder2(new SearchResultDecoder(true), new StringMapInfoDecoder()));

// io.github.dengliming.redismodule.redisearch.protocol.decoder.SearchResultDecoder
public SearchResultDecoder(boolean withScores) {
}


// io.github.dengliming.redismodule.redisearch.RediSearch
return commandExecutor.readAsync(getName(), StringCodec.INSTANCE, searchOptions.isWithScores ? FT_SEARCH_WITH_SCORES : FT_SEARCH, args.toArray());
```
Is it just what you want?

Comment (regisoliveira) [reply redisearch/src/main/java/io/github/dengliming/redismodule/redisearch/protocol/decoder/SearchResultDecoder.java:47]: Yes, it's definitely a good approach, but I was worried about creating a new RedisCommand as you said. I'll make the changes and commit it.

**Comment by regisoliveira** [comment]
Done!

**Comment by dengliming** [comment]
@regisoliveira Can you resolve conflicts?

**Review by regisoliveira — COMMENTED** [review]
Checked!

**Comment by regisoliveira** [comment]
> @regisoliveira Can you resolve conflicts?

I don't have write access to your repo to solve conflicts.

**Comment by dengliming** [comment]
@regisoliveira Thanks. That's merged.

**Pull request merged**